python/feathub/processors/local/local_processor.py

```diff
@@ -178,26 +210,55 @@ def materialize_features(
                 timestamp_field=features.timestamp_field,
                 timestamp_format=features.timestamp_format,
             )
-        elif isinstance(sink, FileSystemSink):
+        elif isinstance(sink, FileSystemSink) and _is_local_file_or_dir(sink.path):
             insert_into_file_sink(features_df, sink)
             return LocalJob()
+        elif is_spark_supported_sink(type(sink)):
+            try:
+                self._init_spark_session_local_mode()
+                return self._materialize_dataframe_using_spark(
+                    df=features_df,
+                    features=features,
+                    sink=sink,
+                    allow_overwrite=allow_overwrite,
+                )
+            except ImportError:
```

Comment:
Would it be simpler to move this try/catch into `_init_spark_session_local_mode`?

Reply:
`_init_spark_session_local_mode` does not know why it is invoked. If we move this try catch into `_init_spark_session_local_mode`, the error message would be changed from 
> Please install Feathub with Spark to use {sink} in LocalProcessor.

to 

> Please install Feathub with Spark to use certain features like HDFS in LocalProcessor.

Which might cause users not knowing clearly which components require Spark.

Reply:
We should be able to use the same error message after moving try/catch to `_init_spark_session_local_mode`, right?

Reply:
Yes, it would simplify the code a bit, but it also makes the error message not show the concrete component that requires Spark.

Reply:
How about moving both try/catch and `_init_spark_session_local_mode` into `_materialize_dataframe_using_spark`?

Same for `_get_table_using_spark`.